src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -501,6 +513,28 @@ void CoreWorkerDirectTaskReceiver::RunNormalTasksFromQueue() {
   normal_scheduling_queue_->ScheduleRequests();
 }
 
+void CoreWorkerDirectTaskReceiver::HandleStealTasks(
+    const rpc::StealTasksRequest &request, rpc::StealTasksReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  size_t half = normal_scheduling_queue_->Size() / 2;
+
+  if (half == 0) {
+    RAY_LOG(DEBUG) << "We don't have enough tasks to steal, so we return early!";
+    reply->set_number_of_tasks_stolen(0);
+    send_reply_callback(Status::OK(), nullptr, nullptr);
+    return;
+  }
+
+  rpc::Address thief_addr = request.thief_addr();
+  size_t n_tasks_stolen = normal_scheduling_queue_->Steal(half, thief_addr, reply);
```

Comment:
We should probably compute `half` inside this method to guard against race conditions.

Reply:
Ok, let me change this!